todoapp/src/components/EditTodoModal.tsx

```diff
@@ -0,0 +1,97 @@
+import { useEffect } from 'react';
+import { Box, Modal, TextField, Button } from '@mui/material';
+import { makeStyles } from '@mui/styles';
+import { useGlobalState } from 'store/TodoStore';
+import { useHookstate } from '@hookstate/core';
+
+const useStyles = makeStyles({
+  popUp: {
+    "&&": {
+      position: 'absolute',
+      top: '50%',
+      left: '50%',
+      transform: 'translate(-50%, -50%)',
+      backgroundColor: 'white',
+      width: '300px',
+      p: '1rem',
+      borderRadius: '8px',
+      padding: 15
+    }
+  },
+  popUpButton: {
+    "&&": {
+      width: '100%',
+      backgroundColor: '#ff5252',
+      color: 'white',
+      '&:hover': {
+        backgroundColor: 'black',
+        color: 'white',
+      },
+    }
+  },
+})
+
+type TodoModalProps = {
+  todoId: string | null;
+  handleCloseModal: () => void;
+};
+
+const TodoModal = ({ todoId, handleCloseModal }: TodoModalProps) => {
+
+  const classes = useStyles();
+  const { tasksList } = useGlobalState();
```

Comment:
IF we do this here : const tasksList = useGlobalState().tasksList.get({ noproxy: true });

we wont have to do  tasksList.value everywhere in this file

Reply:
But when using set() we would still have to access the store using useGlobalState() which is giving an error if used inside the handle change function.
      useGlobalState().tasksList.set(todos);

Reply:
Got it